### PR TITLE
Feature/group list

### PR DESCRIPTION
**Pull request by Atilla13**
This commit introduces a complete groups feature following the existing repository pattern architecture, including data models, repositories, ViewModels, UI screens, and extensive test coverage.

### Core Implementation (7 files)

**Model Layer**
- _Group.kt_: Defined Group data class with id, name, ownerId, description, memberIds, and eventIds
- Added membersCount as computed property from memberIds.size
- Implemented proper data validation and defaults

**Repository Layer**
- _GroupRepository.kt_: Created repository interface with userGroups(), leaveGroup(), and getGroup() methods
- _GroupRepositoryLocal.kt_: Implemented in-memory repository for local/offline use
- Added addTestGroup() helper method for test data creation with automatic memberIds generation
- Added clearAllGroups() helper method for test isolation
- Implemented all repository interface methods with proper error handling
- _GroupRepositoryFirestore.kt_: Implemented Firestore-backed repository for production use
- Queries groups by userId through memberships collection
- Handles Firestore document mapping with proper null safety
- Implements getGroup() with document reference queries
- _GroupRepositoryProvider.kt_: Singleton provider for dependency injection
- Defaults to Firestore implementation
- Allows repository swapping for testing

**ViewModel Layer**
- _GroupListViewModel.kt_: Manages group list UI state using unidirectional data flow
- Exposes StateFlow<GroupListUIState> for reactive updates
- Implements refreshUIState() for manual data refresh
- Handles error states with clearErrorMsg() functionality
- Uses viewModelScope for coroutine management

**UI Layer**
- _GroupListScreen.kt_: Jetpack Compose screen displaying user's groups
- Shows group cards with name, description, and member count
- Implements empty state when no groups exist
- Floating action button for joining new groups
- More options menu for each group card
- Defined comprehensive test tags for instrumentation testing

### Testing (4 test files)

**Unit Tests**
- _GroupRepositoryLocalTest.kt_: Tests local repository CRUD operations
- Verified userGroups() returns correct data
- Tested leaveGroup() removes groups properly
- Validated getGroup() retrieves by ID correctly

- _GroupRepositoryFirestoreTest.kt_: Tests Firestore repository integration
- Mocked Firestore queries and documents using MockK
- Fixed field mappings to match actual Group model (removed category, fixed membersCount)
- Tested null handling and default values
- Verified special character support

- _GroupListViewModelTest.kt_: Tests ViewModel state management
- Fixed all Group instantiations to use correct model fields (ownerId, memberIds instead of category, membersCount)
- Tested loading, refreshing, and error states
- Verified clearErrorMsg() functionality
- Tested edge cases (empty lists, special characters, many groups)

**End-to-End Tests**
- _JoinMeE2ETest.kt_: Added 5 comprehensive groups E2E tests
- e2e_groupsRepository_storesAndRetrievesData: Basic CRUD verification
- e2e_groupsRepository_leavingGroupRemovesIt: Leave functionality validation
- e2e_groupsRepository_getGroupById: ID-based retrieval testing
- e2e_completeUserJourneyWithGroupsData_works: Integration test with events and navigation
- Added GroupRepositoryLocal setup in test initialization with clearAllGroups()
- Imported necessary testing dependencies (GroupRepositoryProvider, assertions)

### Key Architectural Decisions
- Followed existing repository pattern with interface-based abstraction
- Used StateFlow for reactive UI updates following app conventions
- Implemented test helper methods for simplified test data creation
- Maintained consistency with existing code style (ktfmt formatting)
- E2E tests focus on data layer as UI navigation is not yet implemented

Note: UI navigation integration with Profile screen is marked as TODO for future implementation

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/groups/GroupListScreen.kt [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                  onMoreOptions = { onMoreOptionMenu(group) })
+              Spacer(Modifier.height(12.dp))
+            }
+          }
+    } else {
+      Box(
+          modifier = Modifier.fillMaxSize().padding(pd).testTag(GroupListScreenTestTags.EMPTY),
+          contentAlignment = Alignment.Center) {
+            Text(
+                text = "You are currently not\nassigned to a group…",
```
Comment (BryanDMP) [inline app/src/main/java/com/android/joinme/ui/groups/GroupListScreen.kt]: Consider using separate Text composables instead of \n for better screen reader support

File: app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+   * This private method handles the asynchronous operation of fetching groups, managing loading
+   * states, and catching any errors that occur during the fetch operation. Errors are logged and
+   * exposed via the UI state.
+   */
+  private fun getAllGroups() {
+    viewModelScope.launch {
+      try {
+        _uiState.value = _uiState.value.copy(isLoading = true)
+        val groups = repo.userGroups()
+        _uiState.value = GroupListUIState(groups = groups, isLoading = false)
```
Comment (BryanDMP) [inline app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt]: Use .copy() instead of creating new GroupListUIState to preserve any other state fields

File: app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      } catch (e: Exception) {
+        _uiState.value = GroupListUIState(errorMsg = e.message, isLoading = false)
+      }
```
Comment (BryanDMP) [inline app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt]: Consider moving isLoading = false to a finally block to ensure it's always reset even if an exception occurs

File: app/src/main/java/com/android/joinme/repository/GroupRepositoryFirestore.kt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import com.google.firebase.auth.auth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FieldPath
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlin.String
+import kotlinx.coroutines.tasks.await
+
+const val GROUPS_COLLECTION_PATH = "groups"
+const val MEMBERSHIPS_COLLECTION_PATH = "memberships"
+
```
Comment (BryanDMP) [inline app/src/main/java/com/android/joinme/repository/GroupRepositoryFirestore.kt]: KDoc missing !

File: app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+   * exposed via the UI state.
+   */
+  private fun getAllGroups() {
+    viewModelScope.launch {
+      try {
+        _uiState.value = _uiState.value.copy(isLoading = true)
+        val groups = repo.userGroups()
+        _uiState.value = GroupListUIState(groups = groups, isLoading = false)
+      } catch (e: Exception) {
+        _uiState.value = GroupListUIState(errorMsg = e.message, isLoading = false)
```
Comment (BryanDMP) [inline app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt]: Use .copy() instead of creating new GroupListUIState to preserve any other state fields. Also, errorMsg should be set to e.message ?: "Unknown error" to avoid nullable message

File: app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt (hunk 2 above)
Comment (BryanDMP) [inline app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt]: Use .copy() instead of creating new GroupListUIState to preserve any other state fields

File: app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt (hunk 3 above)
Comment (BryanDMP) [inline app/src/main/java/com/android/joinme/viewmodel/GroupListViewModel.kt]: Consider moving isLoading = false to a finally block to ensure it's always reset even if an exception occurs

**Review by BryanDMP — APPROVED** [review]
Looks good now — all the requested changes have been addressed. The code is clean and consistent and good job for the 84.4% in line coverage, so I’m approving and will proceed with the merge.
